test/e2e/service_test.go

```diff
@@ -81,3 +82,66 @@ func (data *TestData) createClusterIPServiceAndBackendPods(t *testing.T, name st
 
 	return svc, cleanup
 }
+
+// TestNodePortWindows tests NodePort Service on Windows Node. It is a temporary test to replace upstream Kubernetes one:
+// https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/test/e2e/windows/service.go#L42
+// Issue: https://github.com/antrea-io/antrea/issues/2289
+func TestNodePortWindows(t *testing.T) {
+	skipIfNoWindowsNodes(t)
+
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer teardownTest(t, data)
+
+	svcName := "agnhost"
+	svcNode := nodeName(clusterInfo.windowsNodes[0])
+	svc, cleanup := data.createAgnhostServiceAndBackendPods(t, svcName, svcNode, corev1.ServiceTypeNodePort)
+	defer cleanup()
+	t.Logf("%s Service is ready", svcName)
+
+	// Unlike upstream Kubernetes Conformance, here client is on a Linux Node (control-plane Node).
+	clientName := "agnhost-client"
+	require.NoError(t, data.createAgnhostPodOnNode(clientName, nodeName(0)))
+	defer data.deletePodAndWait(defaultTimeout, clientName)
+	_, err = data.podWaitForIPs(defaultTimeout, clientName, testNamespace)
+	require.NoError(t, err)
+
+	nodeIP := clusterInfo.nodes[0].ip
+	nodePort := int(svc.Spec.Ports[0].NodePort)
+	addr := fmt.Sprintf("http://%s:%d", nodeIP, nodePort)
+
+	cmd := append([]string{"curl", "--connect-timeout", "1", "--retry", "5", "--retry-connrefused"}, addr)
+	stdout, stderr, err := data.runCommandFromPod(testNamespace, clientName, agnhostContainerName, cmd)
+	if err != nil {
+		t.Errorf("Error when running command '%s' from Pod '%s', stdout: %s, stderr: %s, error: %v",
+			strings.Join(cmd, " "), clientName, stdout, stderr, err)
+	} else {
+		t.Logf("curl from Pod '%s' to `%s` succeeded", clientName, addr)
```

Comment:
nit: why did you use backticks to quote the address?

Reply:
Sorry, it is a mistake. I have changed it to `'%s'`.